iothub/device/src/Transport/AmqpIot/AmqpIotReceivingLink.cs

```diff
@@ -266,12 +266,12 @@ private void OnTwinChangesReceived(AmqpMessage amqpMessage)
                             };
                         }
 
-                        _onTwinMessageReceived.Invoke(null, correlationId, twinException);
+                        _onTwinMessageReceived.Invoke(null, correlationId, twinException).GetAwaiter().GetResult();
```

Comment:
This is a callback defined by AMQP and it doesn't support async so we risk using `GetAwaiter().GetResult()`.